File: mobile/app/src/main/kotlin/com/mesaflow/client/feature/reservation/ReservationPricing.kt
package com.mesaflow.client.feature.reservation

/**
 * Espejo del calculo de fianza del backend (ver reservation-pricing.ts,
 * usado por CreateRestaurantReservationUseCase): un importe fijo por
 * comensal. Se duplica aqui solo para poder mostrar el precio en la UI
 * antes de enviar la reserva; el backend es quien realmente calcula y
 * cobra el importe final, así que un desajuste aquí nunca permite pagar
 * de menos, solo mostraría un precio incorrecto en pantalla (cubierto por
 * ReservationPricingTest).
 */
const val RESERVATION_DEPOSIT_PER_PERSON_CENTS = 500

fun calculateReservationDepositCents(partySize: Int): Int = partySize * RESERVATION_DEPOSIT_PER_PERSON_CENTS

File: backend/src/restaurants/domain/reservation-pricing.ts
// Fianza (fake, no hay cobro real) para reservar mesa: un importe fijo por
// comensal, para desincentivar reservas fantasma sin implicar un pasarela de
// pago real. Ver FakeReservationPaymentGateway (infra) y
// CreateRestaurantReservationUseCase (aplica el cobro antes de crear la
// reserva).
export const RESERVATION_DEPOSIT_PER_PERSON_CENTS = 500;

export function calculateReservationDepositCents(partySize: number): number {
  return partySize * RESERVATION_DEPOSIT_PER_PERSON_CENTS;
}
